src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs

```diff
@@ -264,9 +278,12 @@ private async ValueTask<FlushResult> FlushAsyncInternal(CancellationToken cancel
                 reg = cancellationToken.UnsafeRegister(state => ((StreamPipeWriter)state).Cancel(), this);
             }
 
-            // Update any buffered data
-            _tail.End += _tailBytesBuffered;
-            _tailBytesBuffered = 0;
+            if (_bytesBuffered > 0)
```

Comment:
Is this correct as _bytesBuffered rather than _tailBytesBuffered?  (I don't know the implementation details, just looked a bit odd that they were different.)

Reply:
```suggestion
            if (_bytesBuffered > 0 && _tailBytesBuffered > 0)
```

Reply:
This could be changed to _tailBytesBuffered. It’ll look clearer